apps/condo/domains/notification/tasks.js

```diff
@@ -1,118 +1,166 @@
+const isEmpty = require('lodash/isEmpty')
+
 const conf = require('@core/config')
 const { createTask } = require('@core/keystone/tasks')
 const { getSchemaCtx } = require('@core/keystone/schema')
 
 const { Message } = require('@condo/domains/notification/utils/serverSchema')
-const isEmpty = require('lodash/isEmpty')
+
+const {
+    SMS_TRANSPORT,
+    EMAIL_TRANSPORT,
+    PUSH_TRANSPORT,
+    MESSAGE_SENDING_STATUS,
+    MESSAGE_RESENDING_STATUS,
+    MESSAGE_PROCESSING_STATUS,
+    MESSAGE_ERROR_STATUS,
+    MESSAGE_DELIVERED_STATUS,
+    TRANSPORT_PRIORITIES_BY_MESSAGE_TYPES,
+    DEFAULT_TRANSPORT_PRIORITIES,
+} = require('./constants/constants')
 const sms = require('./transports/sms')
 const email = require('./transports/email')
-const { SMS_TRANSPORT, EMAIL_TRANSPORT, MESSAGE_SENDING_STATUS, MESSAGE_RESENDING_STATUS, MESSAGE_PROCESSING_STATUS, MESSAGE_ERROR_STATUS, MESSAGE_DELIVERED_STATUS } = require('./constants/constants')
+const push = require('./transports/push')
 
 const SEND_TO_CONSOLE = conf.NOTIFICATION__SEND_ALL_MESSAGES_TO_CONSOLE || false
 const DISABLE_LOGGING = conf.NOTIFICATION__DISABLE_LOGGING || false
 
 const TRANSPORTS = {
     [SMS_TRANSPORT]: sms,
     [EMAIL_TRANSPORT]: email,
+    [PUSH_TRANSPORT]: push,
 }
 
 async function _sendMessageByAdapter (transport, adapter, messageContext) {
-    if (SEND_TO_CONSOLE) {
+    if (SEND_TO_CONSOLE && transport !== PUSH_TRANSPORT) {
         if (!DISABLE_LOGGING) console.info(`MESSAGE by ${transport.toUpperCase()} ADAPTER: ${JSON.stringify(messageContext)}`)
+
         return [true, { fakeAdapter: true }]
     }
+
     return await adapter.send(messageContext)
 }
 
+// TODO(DOMA-2396): Add customizable transportation priorities for different types of notifications
+/**
+ * Calculates transport types queue for a message according to message type, transportation priorities for
+ * message type and fallback hardcoded priorities for edge cases.
+ * @param message
+ * @returns {Promise<*[]>}
+ * @private
+ */
+async function _choseMessageTransport (message) {
```

Comment:
It's look like a wrong logic.

 1) if you have a phone it means that you MUST send it by phone!
 2) if you have an email it means that you MUST send this message by email.
 3) if you have a just user it means you can choose user-preferred transport or just prefered transport by system